src/main/java/ga/rugal/food/springmvc/controller/MenuAction.java

```diff
@@ -42,22 +45,32 @@
 
     @Autowired
     private MenuService menuService;
+    
+    @Autowired
+    private RestaurantService restaurantService;
 
+    /**
+     * Get menu information through URL /menu
+     * This method is to get a random menu from a menu list associated with specific restaurant
+     * 
+     * @return Give successful message and menu data in JSON format if the menu exist, 
+     * on the contrary, return failed message if menu can't be found
+     */
     @ResponseBody
     @RequestMapping(method = RequestMethod.GET)
-    public Message randomMenu()
-    {
-        int total = menuService.countTotal();
-        LOG.debug(CommonLogContent.MENU_NUMBER, total);
-        if (0 == total)
-        {
+    public Message menuByRestaurant() {
+  
+        Restaurant restaurant = restaurantService.getRandomRestaurant();
+        List<Menu> menuList = menuService.getMenusByRestaurant(restaurant);
```

Comment:
This line could be optimized by using count by mid. You know, instead of fetching a lot of data from database, simply count the number of related menus from DB is already well enough.  
Please just replace the `getMenusByRestaurant()` method by something like `countMenuByRestaurant(Restaurant r)`.